spot_driver/spot_driver/spot_ros2.py

```diff
@@ -220,7 +218,7 @@ def __init__(self, parameter_list: Optional[typing.List[Parameter]] = None, **kw
         self.callbacks["lease"] = self.lease_callback
         self.callbacks["world_objects"] = self.world_objects_callback
 
-        self.group: CallbackGroup = ReentrantCallbackGroup()
+        self.group: CallbackGroup = MutuallyExclusiveCallbackGroup()
```

Comment:
I have not thoroughly analyzed the implications of this change yet. If this is too conservative, we may have to synchronize closer to `SpotWrapper` call sites.

Reply:
Circling back to this. I experienced some hangs during testing, which went away after applying https://github.com/bdaiinstitute/spot_ros2/pull/298/commits/7be47914f989c81818d99bd6322d71d733e8d118, which removes callbacks that need not share a callback group from the main, now mutually exclusive, callback group.